packages/teams-js/src/private/files.ts

```diff
@@ -296,6 +297,105 @@ export namespace files {
     extension: string;
   }
 
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Actions specific to 3P cloud storage provider file and / or account
+   */
+  export enum CloudStorageProviderFileAction {
+    New = 'NEW',
+    Rename = 'RENAME',
+    Download = 'DOWNLOAD',
+    Upload = 'UPLOAD',
+    Delete = 'DELETE',
+    Logout = 'LOGOUT',
+    Move = 'MOVE',
+    Copy = 'COPY',
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Interface for 3P cloud storage provider request content type
+   */
+  export interface CloudStorageProviderRequest<T> {
+    content: T;
+  }
+
+  /**
+   * Base interface for 3P cloud storage provider action request content
+   */
+  interface BaseCloudStorageProviderContent {
+    action: CloudStorageProviderFileAction;
+    providerCode: CloudStorageProvider;
+  }
+
+  /**
+   * Interface representing 3P cloud storage provider logout action
+   */
+  interface CloudStorageProviderLogoutContent extends BaseCloudStorageProviderContent {
+    action: CloudStorageProviderFileAction.Logout;
+  }
+
+  /**
+   * Interface representing 3P cloud storage provider add new file action
+   */
+  interface CloudStorageProviderNewFileContent extends BaseCloudStorageProviderContent {
+    action: CloudStorageProviderFileAction.New;
+    newFileName: string;
+    newFileType: string;
```

Comment:
Is this file extension? If so might make more sense to name it something like "newFileExtension". Are they supposed to prepend it with a '.'? So, '.doc' vs. 'doc'? Might be worth documenting that

Reply:
Yes it is file extension. Renamed it and added the relevant documentation.